src/suggestion.rs

```diff
@@ -135,6 +145,67 @@ impl<'s> fmt::Display for Suggestion<'s> {
             0
         };
 
+        // We will trim the literal if does not fit in the terminal size
+        // The misspelled word shall always be shown and we aim to show as much
+        // info as possible to the user easily locate the word
+        let terminal_size: usize = get_terminal_size();
+        let mut max_chars: usize = terminal_size;
+        let mut min_chars = 0usize;
+        // The paddings give some space for the ` {} ...` and extra indentation and formatting:
+        // ```
+        // 65 |  ...  here, on the second most pointless site ever! Well, wander ont  ...
+        //    |                                                                  ^^^
+        //    |                                                                                                                                                                                                ^^^
+        //    | - not, Ont, int, on, onto, cont, font, or one of 9 others
+        //    |
+        //    |   Possible spelling mistake found.
+        //    |
+        // ```
+        let padding_offset = 5;
+        let padding_end = 15usize;
+
+        let range_start_text = Range {
+            start: 0usize,
+            end: marker_range_relative.start,
+        };
+        let range_end_text = Range {
+            start: marker_range_relative.end,
+            end: self.literal.as_str().chars().count(),
+        };
+        if self.literal.as_str().chars().count() > terminal_size {
+            // Attempt to fit the misspelled word in the beginning followed by info.
+            if range_end_text.len() >= terminal_size {
+                min_chars = marker_range_relative.start - 1;
+                max_chars = (min_chars + terminal_size).saturating_sub(padding_end);
+            }
+            // Attempt to fit the misspelled word in the end after the info.
+            else if range_start_text.len() > terminal_size {
+                min_chars = (marker_range_relative.end)
+                    .saturating_sub(terminal_size)
+                    .saturating_add(padding_end);
+                max_chars = marker_range_relative.end;
```

Comment:
I see that you create two windows here, one starting from the beginning and one starting from the end.
But if the case we are dealing with is extra long 
```text
veeeeeeeeeeeeeeeeeeeeeery lang mistaaaaaaaaaaaaaaaaake
                                                  ^^^^
|----window-----| 
                                                                               |----window-----| 
```
we might not get the mistake, because the assumption that the windows overlap is not true, note that that is also the case for small terminal sizes.

What do you think about taking the range and based on that extracting and extended section?

I might be entirely wrong here, so feel free to smarten me up :)

Reply:
Could you please elaborate a bit on this or maybe just run an example? I could not reproduce it an error when this is happening mainly because one misspelled is done per time.

Maybe if I see with the current code, I will be able to fix or think in a proper solution.

Reply:
Yeah, alignment above went to s*it (fixed).

I think with the current impl the above is obsolete :+1: